server/sonar-server/src/main/java/org/sonar/server/ws/JsonWriterUtils.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.ws;
+
+import java.util.Collection;
+import javax.annotation.Nullable;
+import org.sonar.api.utils.text.JsonWriter;
+
+public class JsonWriterUtils {
+
+  public static void writeIfNeeded(JsonWriter json, @Nullable String value, String field, Collection<String> fields) {
+    if (isFieldWanted(field, fields)) {
+      json.prop(field, value);
+    }
+  }
+
+  public static void writeIfNeeded(JsonWriter json, @Nullable Boolean value, String field, Collection<String> fields) {
+    if (isFieldWanted(field, fields)) {
+      json.prop(field, value);
+    }
+  }
+
+  public static void writeIfNeeded(JsonWriter json, @Nullable Integer value, String field, Collection<String> fields) {
+    if (isFieldWanted(field, fields)) {
+      json.prop(field, value);
+    }
+  }
+
+  public static void writeIfNeeded(JsonWriter json, @Nullable Long value, String field, Collection<String> fields) {
+    if (isFieldWanted(field, fields)) {
+      json.prop(field, value);
+    }
+  }
+
+  public static boolean isFieldWanted(String field, @Nullable Collection<String> fields) {
```

Comment:
isFieldNeeded to be consistent in naming. I'm not convinced with the writeIfNeeded, but good enough